core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -293,6 +294,8 @@ public void commit() {
             taskOps.commit(base, updated.withUUID());
           });
 
```

Comment:
Fixes deletion of data files, manifests and snapshots related to unknown state commits